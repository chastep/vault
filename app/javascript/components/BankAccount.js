import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const BankAccount = ({ bankAcct, onDelete }) => (
  <div>
    <h2>
      {bankAcct.nickname + ' - ' + bankAcct.created_at}
      <Link to={`/bank_accounts/${bankAcct.id}/edit`}>Edit</Link>
      <button className="delete" type="button" onClick={() => onDelete(bankAcct.id)}>
        Delete
      </button>
    </h2>
    <ul>
      <li>
        <strong>Account Number: </strong>
        {bankAcct.account_number}
      </li>
      <li>
        <strong>Routing Number: </strong>
        {bankAcct.routing_number}
      </li>
      <li>
        <strong>City: </strong>
        {bankAcct.city}
      </li>
    </ul>
  </div>
);

BankAccount.propTypes = {
  bankAcct: PropTypes.shape(),
  onDelete: PropTypes.func.isRequired,
};

BankAccount.defaultProps = {
  bankAcct: undefined,
};

export default BankAccount;
