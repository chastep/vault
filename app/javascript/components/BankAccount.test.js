import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom'

import BankAccount from './BankAccount';

it('renders', () => {
  const deleteFunc = jest.fn();
  const bankAcct = {
    id: 1,
    account_number: '1231231230',
    routing_number: '011000015',
    nickname: 'Test Account',
    location_id: 1,
    address: '123 Fake St',
    address2: 'Apt 4',
    city: 'Chicago',
    region: 'IL',
    postal: '60606',
    created_at: '2019-03-26T18:10:38.029Z',
    updated_at: '2019-03-26T18:10:38.029Z'
  };

  const component = renderer.create(
    <MemoryRouter>
      <BankAccount onDelete={deleteFunc} bankAcct={bankAcct} />
    </MemoryRouter>
  ).toJSON();
  expect.assertions(1);
  expect(component).toMatchSnapshot();
});
