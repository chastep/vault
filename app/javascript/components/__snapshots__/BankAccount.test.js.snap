// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <h2
    className="ui left floated header"
  >
    Bank Account Details - 
    Test Account
  </h2>
  <h3
    className="ui right floated header"
  >
    <a
      href="/bank_accounts/1/edit"
      onClick={[Function]}
    >
      Edit
    </a>
  </h3>
  <table
    className="ui celled table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
        >
          Nickname
        </th>
        <th
          className=""
        >
          Account Number
        </th>
        <th
          className=""
        >
          Routing Number
        </th>
        <th
          className=""
        >
          Address
        </th>
        <th
          className=""
        >
          Created At
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className=""
      >
        <td
          className="single line"
        >
          <a
            href="/bank_accounts/1"
            onClick={[Function]}
          >
            Test Account
          </a>
        </td>
        <td
          className=""
        >
          1231231230
        </td>
        <td
          className=""
        >
          011000015
        </td>
        <td
          className=""
        >
          123 Fake St, Apt 4 Chicago, IL 60606
        </td>
        <td
          className="single line"
        >
          03/26/2019 01:10 PM
        </td>
      </tr>
    </tbody>
  </table>
  <button
    className="ui button delete"
    onClick={[Function]}
    type="button"
  >
    Delete
  </button>
</div>
`;
